### ISSUE TITLE
Build Init plugin should create project files only when all questions are answered

### ISSUE THREAD

**Issue by lacasseio** [gradle/gradle#9534]
### Expected Behavior

Cancelling the init process (via ctrl-c) will leave wrapper and build script behind.

### Current Behavior

The build init process is eager and will create the wrapper (by depending on the wrapper task) and update the build script as the questions are answered.

**Comment by lacasseio** [comment]
The init task should invoke the wrapper generation implementation instead of depending on the wrapper task to allow cleanup.

**Comment by dean-burdaky** [comment]
I've taken a look at the issue and code on the master branch (I assume that the issue isn't stale).

I suggest that the BuildInit class be separated into its own task like "initLayout" since it seems to only generate the layout, and have a "wrapper must come after initLayout" relation. Then the init task can be an empty/null task (or if there is a special composite task type) that depends on both the wrapper and initLayout tasks.

This allows build layout generation to be invoked separately and would solve the issue above. One caveat is that a .gradle directory (and its containing files and directories) will still be generated before the questions are answered but these files and directories are generated first for all tasks invoked.

In terms of testing, some of the existing tests will need tweaking to account for the new task and structure. I also propose an additional test to be created to ensure that no files (bar the .gradle directory) are created before the last question is answered. I assume there is already a test that ensures that all the files are created once the init task is complete.

**Comment by dean-burdaky** [comment]
Actually, the problem could be simply solved by changing dependsOn with finalizedBy on line 56 of the BuildInitPlugin class. Only one test needs to be changed, which is the "applies plugin" test in BuildInitPluginSpec to account for the use of finalizedBy rather than dependsOn.

**Comment by ov7a** [comment]
Sorry for the late reply.

This is a duplicate of
* https://github.com/gradle/gradle/issues/18080
And it was fixed in 8.6 via
* https://github.com/gradle/gradle/pull/27408
